### PR TITLE
Remove legacy OODT dependencies

### LINKED ISSUE #75: Refactor code to remove OODT dependencies
### 💡 Description

The code is very old and prone to vulnerabilities. Let's refactor it out.

It is primarily used for crawler, metadata extraction and in-memory storage, which I think we can just create objects to do and/or use existing libraries to do it (e.g. validate has it's own crawler code)

### ⚔️ Parent Epic / Related Tickets

_No response_

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary
Refactor to remove legacy OODT dependencies.

NOTE: Merge #79 into this branch before merging to main

## ⚙️ Test Data and/or Report
Still working on branch testing workflow, but with everything deployed locally, I was able to run the smoke tests successfully:

```
Summary:

5 of 5 file(s) processed, 0 other file(s) skipped
0 error(s), 0 warning(s)

Product Labels:
5          Successfully registered
0          Failed to register

Registry Search Solr Documents:
5          Successfully created
0          Failed to get created

Product Types Handled:
2 Product_Collection
2 Product_Context
1 Product_Bundle


Registry Package Id: e3ca72b0-09b5-4ce9-909b-483ea5160867

End of Log

[SUCCESS] Harvest Successful
[INFO] Check solr docs match expected
[SUCCESS] Solr Doc Diff Successful
[INFO] Registry Manager Load
[SUCCESS] Registry Manager Load Successful
```

## ♻️ Related Issues

Resolves #75

### CODE REVIEW

File: src/main/java/gov/nasa/pds/harvest/search/crawler/actions/FileObjectRegistrationAction.java [hunk 1]
```diff
@@ -298,10 +298,11 @@ private Metadata createFileObjectMetadata(FileObject fileObject,
     if (!fileObject.getAccessUrls().isEmpty()) {
       slots.add(new Slot(Constants.ACCESS_URLS, fileObject.getAccessUrls()));
     }
-    for (Iterator i = sourceMet.getHashtable().entrySet().iterator();
-    i.hasNext();) {
-      Map.Entry entry = (Map.Entry) i.next();
-      String key = entry.getKey().toString();
+    // for (Iterator i = sourceMet.getHashTable().entrySet().iterator();
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/harvest/search/crawler/actions/FileObjectRegistrationAction.java:301]: When considering commenting-out code, either:

- Include a comment explaining why it's commented out … or …  🤔
- If you can't come up with one, then just delete the code completely 😏

File: src/main/java/gov/nasa/pds/harvest/search/crawler/actions/StorageIngestAction.java [hunk 2]
```diff
@@ -1,191 +1,185 @@
-package gov.nasa.pds.harvest.search.crawler.actions;
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/harvest/search/crawler/actions/StorageIngestAction.java:1]: Was this entire file commented-out?

Could the file just be deleted?

File: src/main/java/gov/nasa/pds/harvest/search/oodt/crawler/Crawler.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+//// Copyright 2006-2017, by the California Institute of Technology.
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/harvest/search/oodt/crawler/Crawler.java:1]: This looks like a _new_ file but is entirely commented-out … could this just not be part of the commit?

**Review by nutjob4life — CHANGES_REQUESTED** [review]
Whew! What a herculean effort … just a couple minor issues, take a look?

**Comment by jordanpadams** [comment]
@nutjob4life thanks for the review? making these updates on #80 to avoid potential merge conflicts.

**Comment by jordanpadams** [comment]
Cleanup completed here: https://github.com/NASA-PDS/registry-harvest-legacy/commit/22957861539c62473c3dc0af6e23899c8c7df8d7
